app/server/utils/ussd/kenya_ussd_state_machine.py

```diff
@@ -757,14 +776,19 @@ def __init__(self, session: UssdSession, user: User):
         balance_inquiry_pin_authorization_transitions = [
             {'trigger': 'feed_char',
              'source': 'balance_inquiry_pin_authorization',
-             'dest': 'complete',
-             'conditions': 'is_authorized_pin',
-             'after': 'inquire_balance'
+             'dest': 'exit_user_balance',
+             'conditions': ['is_authorized_pin', 'has_token_exchanges']
              },
             {'trigger': 'feed_char',
              'source': 'balance_inquiry_pin_authorization',
-             'dest': 'exit_pin_blocked',
-             'conditions': 'is_blocked_pin'}
+             'dest': 'exit_user_exchange_limit_balance',
+             'conditions': ['is_authorized_pin', 'has_default_limit']
+             },
+            {'trigger': 'feed_char',
+             'source': 'balance_inquiry_pin_authorization',
+             'dest': 'exit_user_exchange_balance',
+             'conditions': 'is_blocked_pin',
+             'unless': ['has_token_exchanges', 'has_default_limit']}
```

Comment:
Why is this `unless` needed? It seems strange that any of those states should override the `is_blocked_pin` state.

Reply:
Slight oversight here, pushing a fix in a second.